appdaemontestframework/hass_mocks.py

```diff
@@ -1,12 +1,32 @@
 import logging
 import mock
 from appdaemon.plugins.hass.hassapi import Hass
+import appdaemon.utils
+from packaging import version
 import textwrap
 import warnings
 
 
+class _DeprecatedAppdaemonWarning:
+    # Use class vars so we can keep sticky state to only warn once per test session
+    already_warned_deprecated_appdaemon_version = False
+    min_supported_version = version.Version('4.0.0')
+
+    def __init__(self):
+        if self.already_warned_deprecated_appdaemon_version:
+            return
+        self.already_warned_deprecated_appdaemon_version = True
+
+        appdaemon_version = version.Version(appdaemon.utils.__version__)
+        if appdaemon_version < self.min_supported_version :
+            warnings.warn("Appdaemon-Test-Framework will only support Appdaemon >={} in the next major release. Your current Appdemon version is {}"
+                            .format(self.min_supported_version, appdaemon_version), DeprecationWarning)
+
+
 class HassMocks:
     def __init__(self):
+        _DeprecatedAppdaemonWarning()
```

Comment:
Not entirely sure why we need a class here. But I'll investigate, and try to do away with it. I'll keep you updated.

Reply:
Ok, I see what you were trying to do with the class. Great idea 👍
I think it didn't quite work as expected though, so I just pushed a fix. Let me know if that's what you had in mind.
This commit only fixes the behavior, I'll  do the proposed refactor in separate commits so it's not confusing

Reply:
Yeah, it looks like a jumbled the access to the class variable. I think you got it, but the intent of using a class is to scope the global state of if we have already warned inside of the class rather that just having some free global variables which feels a bit messy.

Reply:
Yep perfect! 
I'll do some minor readability refactors, merge the 2 places where we check for appdaemon version, and I'll push that :)